### PR TITLE
Complete Environment Setup with README and Code Fixes.

### PR DESCRIPTION
**Pull request by Mvamd12**
This PR addresses key updates to ensure the project environment is fully functional, provides comprehensive documentation, and includes minor code fixes to improve stability.

### CODE REVIEW

**Review by Omarmanassrah — APPROVED** [review]

File: app/api/auth_utlis.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 from typing import Optional
-import os
 from fastapi.security import OAuth2PasswordBearer
 from dotenv import load_dotenv
 from passlib.context import CryptContext
+from pydantic import BaseSettings
+
 
 
 load_dotenv()
```
Comment (FaresAbuIram) [inline app/api/auth_utlis.py]: Not sure if we still need this?

File: app/api/routes/login.py [hunk 2]
```diff
@@ -1,33 +1,42 @@
...
+def login(
+    form_data: OAuth2PasswordRequestForm = Depends(),
+    db: Session = Depends(get_db)
+):
+    # Fetch the user by username
+    user = user_service.get_user_by_username(form_data.username, db)
+    if not user:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Incorrect username or password",
```
Comment (FaresAbuIram) [inline app/api/routes/login.py]: Can you please change the message to smth like User doesn't exist

File: app/api/routes/login.py [hunk 3]
```diff
@@ -1,33 +1,42 @@
...
-async def login_for_access_token(form_data: OAuth2PasswordRequestForm = Depends()):
-    user = await get_user_by_username(form_data.username)
-
-# Verify the provided password against the stored hashed password
-    if not user or not pwd_context.verify(form_data.password, get_password_hash(user)):
+    # Verify password
+    if not verify_password(form_data.password, user.hashed_password):
         raise HTTPException(
             status_code=status.HTTP_401_UNAUTHORIZED,
             detail="Incorrect username or password",
```
Comment (FaresAbuIram) [inline app/api/routes/login.py]: The same for password

**Review by FaresAbuIram — APPROVED** [review]
Well done
Please address the suggestions before merging it

**Pull request merged**